### PR TITLE
End-to-end tests

### PR DESCRIPTION
**Pull request by johanneswuerbach**
End-to-end test for the provider

The logging part will only pass once https://github.com/ofiliz/virtual-kubelet/pull/1 and https://github.com/ofiliz/virtual-kubelet/pull/3 are merged.

Currently the tests are failing with:
```
2018/04/21 15:04:08 RegisterTaskDefinition input:{
  ContainerDefinitions: [{
      Command: ["-c","echo \"Started\"; while true; do sleep 1; done"],
      Cpu: 204,
      EntryPoint: ["/bin/sh"],
      Image: "busybox",
      LogConfiguration: {
        LogDriver: "awslogs",
        Options: {
          awslogs-region: "us-east-1",
          awslogs-stream-prefix: "vk-podspec_arn:aws:ecs:us-east-1:439744694103:cluster/vk-aws-e2e-test_default_test_1524315848382468_echo-container",
          awslogs-group: "/ecs/vk-aws-e2e-test"
        }
      },
      Memory: 450,
      MemoryReservation: 256,
      Name: "echo-container"
    }],
  Cpu: "256",
  Family: "vk-podspec_arn:aws:ecs:us-east-1:439744694103:cluster/vk-aws-e2e-test_default_test_1524315848382468",
  Memory: "512",
  NetworkMode: "awsvpc",
  RequiresCompatibilities: ["FARGATE"]
}
2018/04/21 15:04:08 RegisterTaskDefinition err:ClientException: Family contains invalid characters.
	status code: 400, request id: 7a38f81a-4564-11e8-957f-f35d2c658fc8 output:{

}
```

Are those the not-yet released changes you mentioned @ofiliz?

### CODE REVIEW

**Comment by ofiliz** [comment]
No. The RegisterTaskDefinition call is failing because cluster ARN contains characters ':' and '/'. Family field can only consist of alphanumeric, hypen and underscore.

We'll either have to remove cluster name from the family, or use the cluster name (not the ARN).

I wanted the cluster name in the family, because task definitions are shared between clusters and we should be able to tell which task def is created for which cluster.

File: providers/aws/test_helpers.go [hunk 1]
```diff
@@ -0,0 +1,284 @@
+package aws
```
Comment (ofiliz) [inline providers/aws/test_helpers.go]: Any reason why this file is called test_helpers.go? Not having a _test.go extension means this file will be compiled with (although probably not linked into) the actual virtual kubelet binary. helpers_test.go would be a better name.

Were you planning to call these functions from any other file than provider_test.go? If not, should we just move these functions to the end of provider_test.go?

Comment (johanneswuerbach) [reply providers/aws/test_helpers.go]: I can just include them, but was worried that the provider_test would quickly grow in size especially when adding other tests. `helpers_test.go` would work for me.

Comment (ofiliz) [reply providers/aws/test_helpers.go]: Sounds good.

**Comment by johanneswuerbach** [comment]
> We'll either have to remove cluster name from the family, or use the cluster name (not the ARN).

Sounds good, do you want to implement this change or should I? Happy to try :-)

File: providers/aws/test_helpers.go (hunk 1 above)
Comment (johanneswuerbach) [reply providers/aws/test_helpers.go]: Changed 👍

**Pull request merged**